base/int.jl

```diff
@@ -753,6 +757,15 @@ julia> big"7891.5"
 julia> big"_"
 ERROR: ArgumentError: invalid number format _ for BigInt or BigFloat
 [...]
+
+julia> foo() = big"1"
+foo (generic function with 1 method)
+
+# `x` and `y` point to the same BigInt instance
+julia> let x = foo(), y = foo()
+           x === y
+       end
+true
```

Comment:
Maybe a warning is warranted here too? But remember to rebase since #50800 got merged recently.